src/Main.hs

```diff
@@ -0,0 +1,80 @@
+module Main (main) where
+
+import Parser (Expr(..), parseExpr, parseNumber)
+import Evaluator
+import Memory
+import Text.Parsec (parse, ParseError, (<|>), try, string, spaces, char, many1, letter)
+import Text.Parsec.String (Parser)
+import qualified Data.Map as Map
+
+
+-- command type to handle different user inputs
+data Command
+    = ExprCommand Expr        -- mathematical expression
+    | MemorySet String Double -- set a memory value
```

Comment:
Scientific calculators don't usually allow you to name variables. They only store one number in the memory and provide operations to clear it or update in several ways (m+, m-, mc, mr, ...)

Reply:
Should I leave it as a feature or remove it?